safe/impact_function_v4/test/test_impact_function.py

```diff
@@ -381,6 +390,59 @@ def test_post_processor(self):
                 self.assertIn(field_name, impact_fields)
         print_attribute_table(impact_layer, 1)
 
+    def test_provenance(self):
+        """Test provenance of impact function."""
+        hazard_layer = load_test_vector_layer(
+            'gisv4', 'hazard', 'classified_vector.geojson')
+        exposure_layer = load_test_vector_layer(
+            'gisv4', 'exposure', 'building-points.geojson')
+        aggregation_layer = load_test_vector_layer(
+            'gisv4', 'aggregation', 'small_grid.geojson')
+
+        expected_provenance = {
+            'aggregation_keywords': deepcopy(aggregation_layer.keywords),
+            'aggregation_layer': aggregation_layer.source(),
+            'exposure_keywords': deepcopy(exposure_layer.keywords),
+            'exposure_layer': exposure_layer.source(),
+            'hazard_keywords': deepcopy(hazard_layer.keywords),
+            'hazard_layer': hazard_layer.source(),
+        }
+
+        # Set up impact function
+        impact_function = ImpactFunction()
+        impact_function.aggregation = aggregation_layer
+        impact_function.exposure = exposure_layer
+        impact_function.hazard = hazard_layer
+        status, message = impact_function.run()
+        self.assertEqual(ANALYSIS_FAILED_BAD_INPUT, status, message)
+        impact_function.prepare()
+        status, message = impact_function.run()
+        self.assertEqual(ANALYSIS_SUCCESS, status, message)
+
+        provenance = impact_function.provenance
+        # from pprint import pprint
+        # pprint(provenance)
+        self.maxDiff = None
+
+        expected_provenance.update({
+            'action_checklist': impact_function.action_checklist(),
+            'analysis_extent': impact_function.analysis_extent.exportToWkt(),
+            'gdal_version': gdal.__version__,
```

Comment:
hi @Gustry I already did it in https://github.com/inasafe/inasafe/pull/3530/files#diff-6d55c27a3cd8f7d2b3d4fac87ede6569R162.
I keep the test like that since I use it to compare. Or did I miss understand your comment?